### PR TITLE
bring in nems integration modifications

### PR DESCRIPTION
**Pull request by mvertens**
Change Description: bring in nems integration modifications

This PR brings in changes that are needed to run CMEPS in NEMS. 
- There are now only two esmFldsExchange files - esmFldsExchange_cesm.F90 and esmFldsExchange_nems.F90
- The fd.yaml file has been separated into fd_nems.yaml and fd_cesm.yaml.
- There is a new attribute coupling_mode that needs to be set to either nems_orig, nems_orig_data or cesm - but default it is cesm
- The clock initialization routines that were in med_time_mod.F90 - have now been moved to drivers/cime/esm_time_mod.F90. This is driver specific and does not belong in the mediator code.

Testing Completed: testlist_drv.xml
All results were bfb with the feb08 baseline EXCEPT for 

  - ERS_Vnuopc_Ln5.f19_f19_mg17.F2000Nuopc.cheyenne_intel.cam-nuopc_cap
  - SMS_Vnuopc_Ln5.f19_f19_mg17.F2000Nuopc.cheyenne_intel.cam-nuopc_cap

These two tests were repeatedly run and sometimes reproduced the feb08 nuopc baselines and sometimes did not. The results that did not reproduce the baselines were consistent with each other. This will be investigated in future tests - but the decision was to bring the CMEPS branch up to master even though these tests were not passing baseline.

app_cesm-cmeps hash: b120695
CIME HASH: c9df0bfd7

### CODE REVIEW

File: cime_config/buildnml [hunk 1]
```diff
@@ -476,11 +476,11 @@ def buildnml(case, caseroot, component):
     for filename in glob.glob(os.path.join(confdir, "*modelio*")):
         shutil.copy(filename, rundir)
 
-    # copy fd.yaml to rundir
+    # copy fd_cesm.yaml to rundir
     cimeroot = case.get_value("CIMEROOT")
     fd_dir = os.path.join(cimeroot, "src","drivers","nuopc","mediator")
-    filename = os.path.join(fd_dir,"fd.yaml")
-    shutil.copy(filename, rundir)
+    filename = os.path.join(fd_dir,"fd_cesm.yaml")
```
Comment (uturuncoglu) [inline cime_config/buildnml]: I think it is better to make this part as generic as possible and it helps to adapt s2s application easily in the near future.

coupling_mode = case.get_value('coupling_mode')
if coupling_mode == 'cesm':
     filename = os.path.join(fd_dir,"fd_cesm.yaml")
else:
    filename = os.path.join(fd_dir,"fd_nems.yaml")

**Comment by mvertens** [comment]
Good point. I will make that change and push things back.

File: mediator/med_phases_aofluxes_mod.F90 [hunk 2]
```diff
@@ -429,17 +430,29 @@ subroutine med_aofluxes_run(gcomp, aoflux, rc)
...
+       else
+          flux_max_iteration = 1
+       end if
 
-       call NUOPC_CompAttributeGet(gcomp, name='flux_convergence', value=cvalue, rc=rc)
+       call NUOPC_CompAttributeGet(gcomp, name='flux_convergence', value=cvalue, isPresent=isPresent, isSet=isSet, rc=rc)
        if (chkerr(rc,__LINE__,u_FILE_u)) return
-       read(cvalue,*) flux_convergence
+       if (isPresent .and. isSet) then
+          read(cvalue,*) flux_convergence
```
Comment (uturuncoglu) [inline mediator/med_phases_aofluxes_mod.F90]: The default value of flux_convergence is 0.01 in namelist_definition_drv.xml, I think it is better to set it 0.01 if it is not set.

File: mediator/med_phases_aofluxes_mod.F90 [hunk 3]
```diff
@@ -429,17 +430,29 @@ subroutine med_aofluxes_run(gcomp, aoflux, rc)
...
+       end if
 
-       call NUOPC_CompAttributeGet(gcomp, name='flux_max_iteration', value=cvalue, rc=rc)
+       call NUOPC_CompAttributeGet(gcomp, name='flux_max_iteration', value=cvalue, isPresent=isPresent, isSet=isSet, rc=rc)
        if (chkerr(rc,__LINE__,u_FILE_u)) return
-       read(cvalue,*) flux_max_iteration
+       if (isPresent .and. isSet) then
+          read(cvalue,*) flux_max_iteration
+       else
+          flux_max_iteration = 1
```
Comment (uturuncoglu) [inline mediator/med_phases_aofluxes_mod.F90]: The default value of flux_max_iteration is 5 by default. Why we are setting 1 explicitly in here?

File: mediator/med_phases_aofluxes_mod.F90 [hunk 4]
```diff
@@ -429,17 +430,29 @@ subroutine med_aofluxes_run(gcomp, aoflux, rc)
...
 
     if (first_call) then
-       call NUOPC_CompAttributeGet(gcomp, name='coldair_outbreak_mod', value=cvalue, rc=rc)
+       call NUOPC_CompAttributeGet(gcomp, name='coldair_outbreak_mod', value=cvalue, isPresent=isPresent, isSet=isSet, rc=rc)
        if (chkerr(rc,__LINE__,u_FILE_u)) return
-       read(cvalue,*) coldair_outbreak_mod
+       if (isPresent .and. isSet) then
+          read(cvalue,*) coldair_outbreak_mod
+       else
+          coldair_outbreak_mod = .false.
```
Comment (uturuncoglu) [inline mediator/med_phases_aofluxes_mod.F90]: The default is true in namelist_definition_drv.xml.

**Comment by mvertens** [comment]
The reason for the default settings is so that nems can work out of the box
without having to have these values in the config file. The default
settings for the namelist will be used for cesm usage - and you should
always set those for cesm. I just wanted to remove these completely for
nems.

**Comment by uturuncoglu** [comment]
I ran the test suite using following command 

```
qcmd -l walltime=4:00:00 -- ./create_test --xml-testlist ../src/drivers/nuopc/cime_config/testdefs/testlist_drv.xml --xml-machine cheyenne --xml-category nuopc --compare feb08 --baseline-root /glade/p/cesmdata/cseg/nuopc_baselines
```

and I had following failed tests

```
  ERS_Vnuopc_Ld5.T62_g17.C.cheyenne_intel (Overall: FAIL) details:
    FAIL ERS_Vnuopc_Ld5.T62_g17.C.cheyenne_intel COMPARE_base_rest
    FAIL ERS_Vnuopc_Ld5.T62_g17.C.cheyenne_intel BASELINE exception
  ERS_Vnuopc_Ld5.T62_g17.G.cheyenne_intel (Overall: FAIL) details:
    FAIL ERS_Vnuopc_Ld5.T62_g17.G.cheyenne_intel COMPARE_base_rest
    FAIL ERS_Vnuopc_Ld5.T62_g17.G.cheyenne_intel BASELINE exception 
```

Which are different from the test that is listed in the PR. The listed test are passed in my case. 

```
/glade/scratch/turuncu/ERS_Vnuopc_Ln5.f19_f19_mg17.F2000Nuopc.cheyenne_intel.cam-nuopc_cap.C.20200309_142636_ol5gm3
/glade/scratch/turuncu/SMS_Vnuopc_Ln5.f19_f19_mg17.F2000Nuopc.cheyenne_intel.cam-nuopc_cap.C.20200309_142636_ol5gm3
```

**Comment by mvertens** [comment]
The C and G tests are expected fails. The only fails are in the following:
ERS_Vnuopc_Ld5.T62_g17.C.cheyenne_intel.GC.mar08.pop.h.once.nc.base.cprnc.out:          of which      2 had non-zero differences

**Pull request merged**